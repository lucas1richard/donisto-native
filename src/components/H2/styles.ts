import { StyleSheet, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultStyles: TextStyle = {
  fontSize: theme.fontSize * 1.7,
  color: theme.black,
  textAlign: 'center'
};

export default StyleSheet.create({
  default: {
    ...defaultStyles
  },
  primary: {
    ...defaultStyles,
    color: theme.primary[500]
  },
  secondary: {
    ...defaultStyles,
    color: theme.secondary[500]
  },
  light: {
    ...defaultStyles,
    color: '#fff'
  }
});
