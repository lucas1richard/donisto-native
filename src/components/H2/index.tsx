import * as React from 'react';
import { Text } from 'react-native';
import styles from './styles';

export interface IH2Props {
  children: string;
  color?: 'primary'|'secondary'|'default'|'light'
}

const H2 = ({ children, color }: IH2Props) => {
  return (
    <Text style={styles[color]}>
      {children}
    </Text>
  );
};

H2.defaultProps = {
  color: 'default'
};

export default H2;
