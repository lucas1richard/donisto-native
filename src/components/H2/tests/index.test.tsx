import * as React from 'react';
import { shallow } from 'enzyme';
import H2, { IH2Props } from '..';

describe('<H2 />', () => {
  const renderComponent = (props: IH2Props) => shallow(<H2 {...props} />);
  it('renders with just text', () => {
    expect(renderComponent({ children: 'mockText' })).toMatchSnapshot();
  });
  it('renders with a color prop', () => {
    expect(renderComponent({
      children: 'mockText',
      color: 'light',
    })).toMatchSnapshot();
  });
});