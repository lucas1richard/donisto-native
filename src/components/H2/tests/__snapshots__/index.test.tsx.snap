// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> renders with a color prop 1`] = `
<Component
  style={
    Object {
      "color": "#fff",
      "fontSize": 34,
      "textAlign": "center",
    }
  }
>
  mockText
</Component>
`;

exports[`<H2 /> renders with just text 1`] = `
<Component
  style={
    Object {
      "color": "#000000",
      "fontSize": 34,
      "textAlign": "center",
    }
  }
>
  mockText
</Component>
`;
